apps/arweave/src/ar_peers.erl

```diff
@@ -300,22 +304,31 @@ discover_peers() ->
 	end.
 
 %% @doc Resolve the domain name of the given peer (if the given peer is an IP address)
-%% and cache it. Return {ok, Peer} | {error, Reason}.
+%% and cache it. Invalidate the cache after ?STORE_RESOLVED_DOMAIN_S seconds.
+%%
+%% Return {ok, Peer} | {error, Reason}.
 resolve_and_cache_peer(RawPeer, Type) ->
 	case ar_util:safe_parse_peer(RawPeer) of
```

Comment:
It looks like this will always try to resolve first, then it inspects the cache. Or am I mis-reading?

Shoudl we check the cache before calling `ar_util:safe_pars_peer`?